unit-tests-java8/src/test/java/org/eclipse/collections/test/bimap/BiMapTestCase.java

```diff
@@ -11,17 +11,101 @@
 package org.eclipse.collections.test.bimap;
 
 import org.eclipse.collections.api.bimap.BiMap;
+import org.eclipse.collections.api.collection.MutableCollection;
 import org.eclipse.collections.api.list.MutableList;
 import org.eclipse.collections.impl.factory.Lists;
 import org.eclipse.collections.test.RichIterableUniqueTestCase;
+import org.eclipse.collections.test.map.MapIterableTestCase;
+import org.junit.Assert;
+import org.junit.Test;
 
 import static org.eclipse.collections.test.IterableTestCase.assertEquals;
 
-public interface BiMapTestCase extends RichIterableUniqueTestCase
+public interface BiMapTestCase extends RichIterableUniqueTestCase, MapIterableTestCase
```

Comment:
BiMaps are MapIterables, so I added the interface here. This causes more diamonds so more methods need to be overridden. In addition, some map tests rely on data where 2 different keys store the same value. This isn't allowed in BiMaps so a few tests are overridden. These assertions could be made even more general with fewer overrides, but this code review is already a lot bigger than I was expecting.